packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -407,16 +582,26 @@ fn array_to_user_errors(array: mv8::Value<'_>) -> Vec<UserError> {
     vec![UserError(fallback)]
 }
 
-fn array_to_user_warnings(array: mv8::Value<'_>) -> Vec<UserWarning> {
+fn array_to_user_warnings<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    array: v8::Local<'s, v8::Value>,
+) -> Vec<UserWarning> {
     // TODO: Extract optional line numbers
-    let fallback = format!("Unparsed: {:?}", array);
-
-    if let mv8::Value::Array(array) = array {
-        let warnings = array
-            .elements()
-            .map(|e: mv8::Result<'_, mv8::Value<'_>>| {
-                e.map(|e| UserWarning {
-                    message: format!("{:?}", e),
+    let fallback = format!("Unparsed: {array:?}");
+
+    if array.is_array() {
+        let array: v8::Local<'s, v8::Array> = array
+            .try_into()
+            .expect("array conversion to never fail as we just checked it was the right type");
+        let warnings = (0..array.length())
+            .map(|i| {
+                let element = array.get_index(scope, i).ok_or_else(|| {
+                    Error::V8(format!(
+                        "Could not get warning at index {i} in array_to_user_warnings"
```

Comment:
Same as above:
Let's not use array_to_user_warnings in an error message, lets say in the UserWarnings array

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/2a832f8dc26357b06320aed8e66cb59822d849a2